replica/compaction_group.hh

```diff
@@ -218,11 +218,42 @@ public:
 using storage_group_vector = utils::chunked_vector<std::unique_ptr<storage_group>>;
 
 class storage_group_manager {
+protected:
+    table& _t;
```

Comment:
I think it belongs more in compaction than in database.

Reply:
Although, maybe not. We abstracted tablets out of compaction with compaction::table_state. Now we're bringing it back. Let's limit the impact of tablets on modules.

We shouldn't just move things around for convenience, we have to think of modularity too.

Reply:
I can keep it in the table's compaction_group_manager implementations, but I thought that the classes here, being defined in replica/ header file, are in the replica database/table module, rather than the compaction module.  I agree they are borderline as they essentially define the border/interface between tables and compaction.

Reply:
I'll do that in the next version.
The tablet_sstable_set doesn't really need a table&, just a schema_ptr (for logging)